File: frontend/app/models/category.ts
import { Model } from 'app/models/model';

export class Category extends Model {

	public name: string = '';
	public description: string = '';
	public color: string = '';
	public _translations: any = {};

	constructor(json: any = {} as Category) {
		super(json.id ? json.id : '');
		this.name = json.name || '';
		this.color = json.color || '';
		this.description = json.description || '';
		this._translations = json._translations && Object.assign(json._translations) || {};
	}

}

File: server/wupportal/src/main/java/de/codeschluss/wupportal/model/Category.java
package de.codeschluss.wupportal.model;

import java.io.Serializable;
import java.util.List;

import javax.persistence.*;

import de.codeschluss.wupportal.activity.ActivityEntity;
import de.codeschluss.wupportal.base.BaseEntity;

/**
 * The persistent class for the categories database table.
 * 
 */
@Entity
@Table(name = "categories")
public class Category extends BaseEntity implements Serializable {
	
	private static final long serialVersionUID = 1L;

	private String color;

	@Lob
	private String description;

	private String name;

	@OneToMany(mappedBy = "category")
	private List<ActivityEntity> activityEntities;

	public Category() {
		super();
	}

	public String getColor() {
		return this.color;
	}

	public void setColor(String color) {
		this.color = color;
	}

	public String getDescription() {
		return this.description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getName() {
		return this.name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public List<ActivityEntity> getActivities() {
		return this.activityEntities;
	}

	public void setActivities(List<ActivityEntity> activityEntities) {
		this.activityEntities = activityEntities;
	}

}